openwillis-face/src/openwillis/face/head_movement.py

```diff
@@ -0,0 +1,396 @@
+        "ZXY": lambda: np.einsum("nij,njk,nkl->nil", R_z, R_x, R_y),  # roll -> pitch -> yaw
+        "ZYX": lambda: np.einsum("nij,njk,nkl->nil", R_z, R_y, R_x),  # roll -> yaw -> pitch
+    }
+
+    if order not in order_map:
+        raise ValueError("Invalid rotation order. Use 'XYZ', 'YXZ', 'ZXY', or 'ZYX'.")
+
+    R = order_map[order]()  # Shape: (N, 3, 3)
+
+    # Compute the total rotation angles from the trace of R
+    trace_R = np.einsum("nii->n", R)  
+    rotation_angles = np.arccos(np.clip((trace_R - 1) / 2, -1.0, 1.0))  
+
+    return np.degrees(rotation_angles)  # Convert to degrees
+
+def head_movement(video_path, frames_per_second=3, normalize_by_bb_size=False, bbox_list=[], padding_percent=0.1):
+    """
+    Extract bounding boxes and facial landmark coordinates from a video using py-feat's Detector,
+    sampling at a specified number of frames per second, and compute various head movement metrics.
+
+    Parameters
+    ----------
+    video_path : str
+        Path to the input video file.
+    frames_per_second : float, optional
+        The number of frames to sample per second of video. Default is 3.
+        This determines the temporal resolution of the analysis.
+    normalize_by_bb_size : bool, optional
+        If True, normalize the xy displacement by the bounding box width. Default is False.
+    bbox_list : list of dict, optional
+        A list of bounding boxes, each represented as a dictionary with keys:
+        ['bb_x1', 'bb_y1', 'w', 'h']. If provided, these bounding boxes will be used 
+        instead of detecting them.
+    padding_percent : float, optional
+        Amount of padding to add around the bounding box as a percentage of its size.
+        Default is 0.1 (10%).
+
+    Returns
+    -------
+    out_df : pd.DataFrame
+        DataFrame containing per-frame results including bounding boxes, landmarks,
+        displacement, and computed rotation metrics.
+    summary_df : pd.DataFrame
+        A one-row DataFrame summarizing mean and std of selected metrics.
+    """
+    
+    out_df = extract_landmarks_and_bboxes(
+        video_path,
+        frames_per_second=frames_per_second,
+        bbox_list=bbox_list,
+        padding_percent=padding_percent
+    )
+    
+    out_df = _compute_bb_centers(out_df)
+
+    sampled_frames = _sample_frames(out_df)
+    sampled_frames = _compute_xy_disp(
+        sampled_frames, 
+        normalize_by_bb_size=normalize_by_bb_size
+    )
+
+    out_df['euclidean_angle'] = compute_rotation_angles_vectorized(
+        out_df['pitch'], 
+        out_df['yaw'], 
+        out_df['roll']
+    )
+    
+    sampled_angles = out_df['euclidean_angle'].dropna()
+    
+    out_df['euclidean_angle_disp'] = sampled_angles.diff().abs()
+
+    out_df['xy_disp'] = sampled_frames['xy_disp']
+
+    summary_df = _compute_summary_stats(out_df)
+
+    return out_df, summary_df
+
+# -------------------------------------------------------------------------
```

Comment:
I would also argue here to remove separating comment for helper functions and bring them to the same format and docstrings as the other functions